alpha/apps/kaltura/lib/kEntitlementUtils.php

```diff
@@ -626,7 +626,7 @@ protected static function validateEntryEntitlements(entry $entry, $kuserId, $ks,
 		// entry that doesn't belong to any category is public
 		//when ks is not provided - the entry is still public (for example - download action)
 		$categoryEntry = categoryEntryPeer::retrieveOneActiveByEntryId($entry->getId());
-		if (!$categoryEntry)
+		if (!$categoryEntry && !PermissionPeer::isValidForPartner(PermissionName::FEATURE_UNCATEGORIZED_ENTRIES_VALIDATION, kCurrentContext::getCurrentPartnerId()))
```

Comment:
if the check is disabled and we have no category entries why even continue to the next checks and not return immediately?